docs/guides/implementation-guides/how-to-implement-dependency-injection.md

```diff
@@ -7,9 +7,120 @@ import MvvmArchitectureDiagram from '/img/gitbook-import/assets/image (3) (1) (2
 
 # How To Implement Dependency Injection
 
-Content in preparation.
+:::info
+For views you can make a simple [ViewLocator](../../concepts/view-locator) which will automatically resolve the view from the Locator.
+:::
 
 This guide will show you how to use Dependency Injection (DI) with _Avalonia UI_ and the MVVM pattern. 
 
 <img src={MvvmArchitectureDiagram} alt=""/>
 
+Let's assume that your MainViewModel as a dependency on MyService. The contructor for ViewModel should looks like this:
+```csharp
+private readonly MyService _myService;
+
+public MapsViewModel(MyService myService)
+{
+    _myService = myService;
+}
+```
+These are the steps you need to do in order to resolve that dependencu using DI.
+
+## Step 1: Install the NuGet package Microsoft.Extensions.DependencyInjection
+```shell
+dotnet add package Microsoft.Extensions.DependencyInjection
+```
+
+## Step 2: ServiceCollectionExtensions 
+The following code is creating an extension for IServiceCollection. This is where you will register all your common dependencies. It's recomended to create that class in the shared project used by all avalonia target platform.
+
+```csharp
+public static class ServiceCollectionExtensions {
+    public static void AddCommonServices(this IServiceCollection collection) {
+        collection.AddTransient<MainViewModel>();
+        collection.AddTransient<MyService>();
+    }
+}
+```
+
+## Step 3: ServiceManager
+The following code create a ServiceManager class that is used to retrieve an instance of a given object.
```

Comment:
There is some incorrect grammar here. It should be "The following code create**s**..."